python/lsst/ts/externalscripts/base_make_calibrations.py

```diff
@@ -0,0 +1,665 @@
+                type: string
+                default: "1950-01-01"
+                descriptor: ISO-8601 datetime (TAI) of the beginning of the \
+                    validity range for the certified calibrations.
+            certify_calib_end_date:
+                type: string
+                default: "2050-01-01"
+                descriptor: ISO-8601 datetime (TAI) of the end of the \
+                    validity range for the certified calibrations.
+            oods_timeout:
+                type: integer
+                default: 120
+                descriptor: Timeout value, in seconds, for OODS.
+        additionalProperties: false
+        """
+        return yaml.safe_load(schema)
+
+    async def set_exp_times_per_im_type(self, image_type):
+        """Define exp_times and n_images according to image type.
+
+        Parameters
+        ----------
+        image_type : `str`
+            Image type. One of ["BIAS", "DARK", "FLAT"].
+        """
+
+        if image_type == "BIAS":
+            n_images = self.config.n_bias
+            exp_times = 0.
+        elif image_type == "DARK":
+            n_images = self.config.n_dark
+            exp_times = self.config.exp_times_dark
+        else:
+            n_images = self.config.n_flat
+            exp_times = self.config.exp_times_flat
+
+        if isinstance(exp_times, collections.abc.Iterable):
+            if n_images is not None:
+                if len(exp_times) != n_images:
+                    raise ValueError(
+                        "n_images_" + f"{image_type}".lower() + f"={n_images} specified and "
+                        "exp_times_" + f"{image_type}".lower() + f"={exp_times} is an array, "
+                        f"but the length does not match the number of images."
+                    )
+        else:
+            # exp_times is a scalar; if n_images is specified then
+            # take that many images, else take 1 image
+            if n_images is None:
+                n_images = 1
+            exp_times = [exp_times] * n_images
+
+        return exp_times
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config: `types.SimpleNamespace`
+            Configuration data. See `get_schema` for information about data
+            structure.
+        """
+        # Log information about the configuration
+
+        self.log.debug(
+            f"n_bias: {config.n_bias}, detectors: {config.detectors}, "
+            f"n_dark: {config.n_dark}, "
+            f"n_flat: {config.n_flat}, "
+            f"instrument: {self.instrument_name} "
+        )
+
+        self.config = config
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script.
+        """
+        n_images = self.config.n_bias + self.config.n_dark + self.config.n_flat
+        metadata.duration = n_images*(self.camera.read_out_time + self.estimated_process_time)
```

Comment:
Is this `metadata.duration` an estimate that causes a timeout, or is it just an estimate?  My concern is that if `n_dark` is 10, and the commanded exposure time is 100s, this will underestimate the real time.  Similarly the flats have non-zero exposure time.

Reply:
I don't think it causes a timeout, it's just an estimate. Is that correct, @tribeiro ?